code/modules/mob/living/carbon/human/species.dm

```diff
@@ -313,103 +313,139 @@ GLOBAL_LIST_EMPTY(features_by_species)
+		var/obj/item/organ/new_organ = get_mutant_organ_type_for_slot(slot)
 
-		var/obj/item/organ/oldorgan = C.getorganslot(slot) //used in removing
-		var/obj/item/organ/neworgan = slot_mutantorgans[slot] //used in adding
-		if(!neworgan) //these can be null, if so we shouldn't regenerate
-			if(oldorgan) // although we also need to remove the old organ if it exists
-				oldorgan.Remove(C, special=TRUE)
-				qdel(oldorgan)
+		if(isnull(new_organ)) // if they aren't suppose to have an organ here, remove it
+			if(existing_organ)
+				existing_organ.Remove(organ_holder, special = TRUE)
+				qdel(existing_organ)
 			continue
 
-		if(visual_only && !initial(neworgan.visual))
+		if(!isnull(old_species) && !isnull(existing_organ))
+			if(existing_organ.type != old_species.get_mutant_organ_type_for_slot(slot))
+				continue // we don't want to remove organs that are not the default for this species
+
+		// at this point we already know new_organ is not null
+		if(existing_organ?.type == new_organ)
+			continue // we don't want to remove organs that are the same as the new one
+
+		if(visual_only && !initial(new_organ.visual))
 			continue
 
 		var/used_neworgan = FALSE
-		neworgan = SSwardrobe.provide_type(neworgan)
-		var/should_have = neworgan.get_availability(src, C) //organ proc that points back to a species trait (so if the species is supposed to have this organ)
-
-		/*
-		 * There is an existing organ in this slot, what should we do with it? Probably remove it!
-		 *
-		 * We will remove it if (and only if):
-		 * - We should not have the organ OR replace_current is passed to force old organs to regenerate
-		 * - The replaced organ is not in an excluded zone
-		 * - The replaced organ is not unremovable or synthetic (an implant)
-		 */
-		if(oldorgan && (!should_have || replace_current) && !(oldorgan.zone in excluded_zones) && !(oldorgan.organ_flags & (ORGAN_UNREMOVABLE|ORGAN_SYNTHETIC)))
-			if(slot == ORGAN_SLOT_BRAIN)
-				var/obj/item/organ/internal/brain/brain = oldorgan
-				if(!brain.decoy_override)//"Just keep it if it's fake" - confucius, probably
-					brain.before_organ_replacement(neworgan)
-					brain.Remove(C, special=TRUE, no_id_transfer=TRUE) //brain argument used so it doesn't cause any... sudden death.
-					QDEL_NULL(brain)
-					oldorgan = null //now deleted
-			else
-				oldorgan.before_organ_replacement(neworgan)
-				oldorgan.Remove(C, special=TRUE)
-				QDEL_NULL(oldorgan) //we cannot just tab this out because we need to skip the deleting if it is a decoy brain.
+		new_organ = SSwardrobe.provide_type(new_organ)
+		var/should_have = new_organ.get_availability(src, organ_holder)
+
+		// Check for an existing organ, and if there is one check to see if we should remove it
+		var/health_pct = 1
+		var/remove_existing = !isnull(existing_organ)
+		if(remove_existing)
+			if(existing_organ.zone in excluded_zones)
+				remove_existing = FALSE
+			else if(existing_organ.organ_flags & ORGAN_UNREMOVABLE)
+				remove_existing = FALSE
```

Comment:
```suggestion
		var/remove_existing = !isnull(existing_organ) && !(existing_organ.zone in excluded_zones) && !(existing_organ.organ_flags & ORGAN_UNREMOVABLE)
```

This line does get pretty long when concatenated but unless you think it presents a seriously problem for readability it's probably better than having the rest of that block indented?

Alternately:
```suggestion
		var/keep_existing = isnull(existing_organ) || (existing_organ.zone in excluded_zones) || (existing_organ.organ_flags & ORGAN_UNREMOVABLE)
```
I find the positive version is often easier to keep in the head than the negative one
Will git freak out if I put two suggestions in one comment? I've never tried it before.

Reply:
For this instance I dislike how putting it all in one statement impacts readability, but at the same time anyone working with species code should understand and be able to read it

Reply:
It seems that I may have tricked you with a slightly incorrect web edit.